File: 069. Sqrt(x)/069. Sqrt(x).java
//二分法搜索

class Solution {
public:
    int mySqrt(int x) {
        if (x<=1) return x;
        int l=0,r=x;
        while(l<=r){
            int mid=l+(r-l)/2;
            if(x/mid>mid) l=mid+1;
            else if(x/mid<mid) r=mid-1;
            else return mid;
        }
        return r;
    }
};

File: 069. Sqrt(x)/069. Sqrt(x).py
'''
思路一
'''
class Solution1(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        r=x
        while r*r>x:
            r=(r+x/r)/2
        return r


#思路二
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x==0 or x==1:
            return x
        i=0
        j=x
        mid=0
        while True:
            mid=(i+j)/2
            if mid>x/mid:
                j=mid
            else:
                if (mid+1)>x/(mid+1):
                    return mid
                i=mid
